### PR TITLE
refactor(client): from fetch api to Axios library

### PR DESCRIPTION
**Pull request by asoran**

### CODE REVIEW

**Review by guyts96 — APPROVED** [review]

**Review by Plantin7 — APPROVED** [review]
💯

File: client/src/services/blocklyService.ts [hunk 1]
```diff
@@ -101,7 +100,7 @@ export default class BlocklyService {
     }
 
     loadAutomation(xml: string): void {
-        BlockFactory.mainWorkspace.clear();
+        Blockly.mainWorkspace.clear();
```
Comment (Lukyio) [inline client/src/services/blocklyService.ts:103]: 👍

File: client/src/services/entities/lightService.ts [hunk 2]
```diff
@@ -1,80 +1,32 @@
...
-    const headers = new Headers();
-    headers.set('Content-Type', 'application/json');
-    fetch(`${config.BASE_URL}/entity/${id}`, {
-        headers,
-        method: 'PUT',
-        body,
-    });
 }
 
 export async function changeBrightness(id: string, brightnessPct: number): Promise<void> {
```
Comment (Lukyio) [inline client/src/services/entities/lightService.ts:10]: can be refactored

File: client/src/services/entities/lightService.ts [hunk 3]
```diff
@@ -1,80 +1,32 @@
...
-    return fetch(`${config.BASE_URL}/entity/${id}`, {
-        headers,
-        method: 'PUT',
-        body,
+    return http.put(`/entity/${id}`, {
+        service: (switchOn ? 'light.turn_on' : 'light.turn_off'),
     });
 }
 
 export async function changeLampRGB(id: string, r: number, g: number, b: number): Promise<void> {
```
Comment (Lukyio) [inline client/src/services/entities/lightService.ts]: in case of

File: client/src/services/entityService.ts [hunk 4]
```diff
@@ -1,53 +1,27 @@
...
-export async function getEntity<T>(id: string): Promise<T> {
-    return fetch(`${config.BASE_URL}/entity/${id}`)
-        .then((x) => {
-            if (!x.ok) {
-                throw Error('');
-            }
-            return x.json();
-        });
+export async function getEntity<T extends Entity<unknown>>(id: string): Promise<T> {
+    return http.get(`/entity/${id}`);
```
Comment (Lukyio) [inline client/src/services/entityService.ts]: 👍

**Pull request merged**